.github/workflows/standard-its.yml

```diff
@@ -100,6 +100,54 @@ jobs:
       override_config_path: ./environment-configs/test-groups/custom-coordinator-duties
       group: custom coordinator duties
 
+  integration-k8s-leadership-tests:
+    name: (Compile=openjdk8, Run=openjdk8, Cluster Build On K8s) ITNestedQueryPushDownTest integration test
+    runs-on: ubuntu-22.04
+    env:
+      MVN: mvn --no-snapshot-updates
+      MAVEN_SKIP: -P skip-static-checks -Dweb.console.skip=true -Dmaven.javadoc.skip=true
+      CONFIG_FILE: k8s_run_config_file.json
+      IT_TEST: -Dit.test=ITNestedQueryPushDownTest
+      POD_NAME: int-test
+      POD_NAMESPACE: default
+      BUILD_DRUID_CLUSTER: true
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v3
+
+      - name: Setup java
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 8
+
+      - name: Restore Maven repository
+        id: maven-restore
+        uses: actions/cache/restore@v3
+        with:
+          path: ~/.m2/repository
+          key: maven-${{ runner.os }}-8-${{ github.sha }}
+
+      - name: Maven build
+        if: steps.maven-restore.outputs.cache-hit != 'true'
+        run: |
+          ./it.sh ci
```

Comment:
`it.sh ci` builds Druid. We'd like to cache builds across stages to avoid load. Building Druid replaces some jars in the Maven cache. Then, the old ITs do its own build, with its own options, replacing the build done here.

The upshot is, I wonder if we need to do this build? Will the old IT run here do it anyway?

Reply:
This is building only when `steps.maven-restore.outputs.cache-hit != 'true'` so it's not building everytime.